fluss-common/src/main/java/org/apache/fluss/metadata/Schema.java

```diff
@@ -303,7 +322,47 @@ public Builder fromColumns(List<Column> inputColumns) {
         public Builder column(String columnName, DataType dataType) {
             checkNotNull(columnName, "Column name must not be null.");
             checkNotNull(dataType, "Data type must not be null.");
-            columns.add(new Column(columnName, dataType, null, highestFieldId.incrementAndGet()));
+            columns.add(
+                    new Column(columnName, dataType, null, highestFieldId.incrementAndGet(), null));
+            return this;
+        }
+
+        /**
+         * Declares a column with aggregation function that is appended to this schema.
+         *
+         * <p>This method associates an aggregation function with a non-primary key column. It is
+         * only applicable when the table uses aggregation merge engine.
+         *
+         * <p>If aggregation function is not specified for a non-primary key column, it defaults to
+         * {@link AggFunction#LAST_VALUE_IGNORE_NULLS}.
+         *
+         * @param columnName the name of the column
+         * @param dataType the data type of the column
+         * @param aggFunction the aggregation function to apply
+         * @return the builder instance
+         * @throws IllegalArgumentException if column is a primary key
+         */
+        public Builder column(String columnName, DataType dataType, AggFunction aggFunction) {
+            checkNotNull(columnName, "Column name must not be null.");
+            checkNotNull(dataType, "Data type must not be null.");
+            checkNotNull(aggFunction, "Aggregation function must not be null.");
+
+            // Validate that this column is not a primary key
+            if (primaryKey != null && primaryKey.columnNames.contains(columnName)) {
+                throw new IllegalArgumentException(
+                        String.format(
+                                "Cannot set aggregation function for primary key column '%s'. "
+                                        + "Primary key columns automatically use 'primary-key' aggregation.",
+                                columnName));
```

Comment:
Could you move the check into `org.apache.fluss.metadata.Schema#normalizeColumns`? That's where we validate the relationship between columns and constraints. Currently, you have to check at multiple places (`column()` and `build()`)